feat: Automatic URL Switching

ariqpradipa opened the pull request:
When users have a Suwayomi server on their local network and also use a public domain, they typically need to manually switch URLs when moving between networks. This implementation automates that process by detecting the current WiFi network and switching between the local and external Suwayomi server URLs accordingly. It enables seamless access without requiring manual changes.

Features
- Network Detection: Automatically detects current WiFi network
- Dynamic Switching: Switches between local and external server URLs based on network
- Local Server Support: Configure local WiFi name and server URL template with dynamic IP resolution
- External URL Management: Add/edit/remove multiple external server URLs with validation
- Server Validation: Validates URLs point to actual Tachidesk/Suwayomi servers
- Manual Override: Users can still manually select servers when needed

ariqpradipa committed:
feat: Add Automatic URL Switching feature with WiFi detection

ariqpradipa committed:
refactor: Remove automatic URL switching providers implementation

ariqpradipa committed:
feat: Implement location permission handling for Automatic URL Switching

ariqpradipa committed:
feat: Enhance automatic URL switching with permission handling and user feedback

ariqpradipa committed:
feat: Update ServerScreen to conditionally display ClientSection based on automatic URL switching status

ariqpradipa committed:
feat: Add section titles and dividers for local and external network configurations in Automatic URL Switching

ariqpradipa committed:
feat: Introduce ExternalUrlConfig for managing external URLs with authentication support

ariqpradipa committed:
feat: Add local network configuration management with UI for adding and editing networks

ariqpradipa committed:
feat: Implement global authentication management with UI integration

ariqpradipa committed:
feat: Enhance Automatic URL Switching with global authentication UI and refactor server URL tile

ariqpradipa committed:
Merge branch 'main' into feat/automatic-url-switching

ariqpradipa committed:
feat: Add Norwegian localization strings for app UI and settings

ariqpradipa committed:
feat: Update iOS deployment target to 12.0 and adjust project settings

ariqpradipa committed:
feat: Add network connection testing functionality and corresponding UI elements

ariqpradipa committed:
feat: Refactor server URL handling to use active server URL with automatic switching logic

ariqpradipa committed:
feat: feat: Implement automatic URL switching in cache manager and server image components

ariqpradipa committed:
feat: Add AppLifecycleManager for automatic URL switching and integrate with Sorayomi widget

ariqpradipa committed:
Merge branch 'main' into feat/automatic-url-switching

ariqpradipa committed:
feat: Add timeout settings for server requests and auto-refresh logic

DattatreyaReddy commented on lib/src/features/settings/presentation/automatic_url_switching/automatic_url_switching_screen.dart:
```diff
@@ -0,0 +1,116 @@
+                  leading: Icon(Icons.info_outline),
+                  title: Text('Network Information'),
+                  subtitle: Text(
+                    'The app will automatically detect your WiFi network and switch to the appropriate server URL.',
```
please use l10n don't hard code string in english

ariqpradipa replied:
sorry, missed this one

DattatreyaReddy commented:
Thank you for your contribution @ariqpradipa 🙇

ariqpradipa committed:
feat: Refactor server URL handling to utilize automatic URL switching

ariqpradipa committed:
feat: Simplify server URL retrieval in GraphQL client with automatic URL switching

ariqpradipa committed:
feat:  separate TestConnectionDialog widget

ariqpradipa committed:
feat: refactor global credentials handling into a separate widget

ariqpradipa committed:
feat: Improve null and empty checks for local and external network configurations

ariqpradipa committed:
Refactor automatic URL switching section by modularizing dialog components

- Added  for adding external URL configurations.
- Added  for adding local network configurations.
- Added  for editing external URL configurations.
- Added  for editing local network configurations.
- Introduced  for displaying external URL configurations.
- Introduced  for displaying local network configurations.
- Added  for requesting permissions.
- Updated  to utilize new dialog and tile components, improving code readability and maintainability.

ariqpradipa committed:
feat: Modularize automatic URL switching section by consolidating dialog components and improving structure
feat: Replace AddExternalUrlDialog and AddLocalNetworkDialog with a unified ExternalUrlDialog and LocalNetworkDialog
feat: Introduce GlobalCredentialsPopup for managing global credentials
feat: Implement TestConnectionDialog for connection testing feedback

ariqpradipa committed:
feat: Rename GlobalCredentialsPopup to GlobalCredentialsDialog and implement dialog functionality

ariqpradipa committed:
feat: Localize text in Automatic URL Switching screen and update ARB file

ariqpradipa committed:
feat: revert timeout handling in GraphQL client

ariqpradipa committed:
feat: enhance ServerUrlTile to support automatic/manual URL switching and implement dialog for URL editing

ariqpradipa committed:
feat: increase connection timeout for more reliable network detection and add timeout handling for server reachability checks

ariqpradipa committed:
feat: implement automatic URL switching logic for server connection parameters

ariqpradipa committed:
feat: enhance URL validation and fallback logic in various components for improved reliability

ariqpradipa committed:
fix: fix automatic URL switching logic and improve error handling for server URL availability
